secure-api-gateway-ig-extensions/src/test/java/com/forgerock/sapi/gateway/scripts/ProcessRegistrationTest.java

```diff
@@ -0,0 +1,672 @@
+import static java.util.Collections.emptyList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.InstanceOfAssertFactories.type;
+import static org.forgerock.http.protocol.Response.newResponsePromise;
+import static org.forgerock.http.protocol.Status.BAD_REQUEST;
+import static org.forgerock.http.protocol.Status.INTERNAL_SERVER_ERROR;
+import static org.forgerock.http.protocol.Status.OK;
+import static org.forgerock.json.JsonValue.array;
+import static org.forgerock.json.JsonValue.field;
+import static org.forgerock.json.JsonValue.json;
+import static org.forgerock.json.JsonValue.object;
+import static org.forgerock.util.promise.Promises.newResultPromise;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import org.forgerock.http.Filter;
+import org.forgerock.http.Handler;
+import org.forgerock.http.protocol.Request;
+import org.forgerock.http.protocol.Response;
+import org.forgerock.json.JsonValue;
+import org.forgerock.json.jose.jwk.JWKSet;
+import org.forgerock.json.jose.jws.SignedJwt;
+import org.forgerock.openig.fapi.apiclient.ApiClient;
+import org.forgerock.openig.fapi.apiclient.ApiClientOrganisation;
+import org.forgerock.openig.fapi.dcr.RegistrationRequest;
+import org.forgerock.openig.fapi.dcr.SoftwareStatement;
+import org.forgerock.openig.fapi.jwks.JwkSetService;
+import org.forgerock.openig.fapi.trusteddirectory.TrustedDirectoryService;
+import org.forgerock.openig.filter.ScriptableFilter;
+import org.forgerock.openig.heap.HeapImpl;
+import org.forgerock.openig.heap.Name;
+import org.forgerock.openig.util.Choice;
+import org.forgerock.services.context.AttributesContext;
+import org.forgerock.services.context.RootContext;
+import org.forgerock.util.Pair;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.KeyUse;
+
+/**
+ * Unit test for {@code ProcessRegistration} script.
+ */
+@ExtendWith(MockitoExtension.class)
+class ProcessRegistrationTest extends AbstractScriptTest {
```

Comment:
This is cool, I didn't know you could do unit tests for IG scripts!!! Is this a documented feature??